File: python3/Q75_sort_colors.py
class Solution:
    def sortColors(self, nums):
        """
        Do not return anything, modify nums in-place instead.
        """
        index = 0
        for i in range(0,len(nums)):
            if nums[i]==1:
                index = i
                break
        temp = nums[0]
        
        nums[0] = nums[index]
        
        nums[index] = temp
        
        lo = 0
        i = lo + 1
        gt = len(nums)-1
        pivot = nums[lo]
        while i <= gt:
            if nums[i] < pivot:
                temp = nums[i]
                nums[i] = nums[lo]
                nums[lo] = temp
                lo += 1
                i += 1
            elif nums[i] > pivot:
                nums[i], nums[gt] = nums[gt], nums[i]
                gt -= 1
                
            else:
                i+=1
                
            
        
            
        
        
        
if __name__ == "__main__":
    arr = [2,0,2,1,1,0]

    s = Solution()
    s.sortColors(arr)

    print(arr)

File: java/src/Q75_Sort_Colors.java
public class Q75_Sort_Colors {

    public void sortColors2Pass(int[] nums) {
        int c0 = 0, c1 = 0, c2 = 0;
        for (int i = 0; i < nums.length; ++i) {
            if (nums[i] == 0) c0++;
            else if (nums[i] == 1) c1++;
            else c2++;
        }

        for (int i = 0; i < nums.length; ++i) {
            if (c0 > 0){
                nums[i] = 0;
                c0--;
            } else if (c1 > 0){
                nums[i] = 1;
                c1--;
            } else {
                nums[i] = 2;
                c2--;
            }
        }
    }

    public void sortColors(int[] nums) {
        int p1 = 0, p2 = nums.length - 1, index = 0;
        while (index <= p2) {
            if (nums[index] == 0) {
                nums[index] = nums[p1];
                nums[p1] = 0;
                p1++;
            }
            if (nums[index] == 2) {
                nums[index] = nums[p2];
                nums[p2] = 2;
                p2--;
                index--;
            }
            index++;
        }

    }

    public static void main(String[] args) {
        Q75_Sort_Colors obj = new Q75_Sort_Colors();
        int[] nums = new int[]{2,0,2,1,1,0};
        obj.sortColors(nums);
        for (int i = 0; i < nums.length; ++i) {
            System.out.print(nums[i] + " ");
        }
    }
}
